source/spawn/spawn-processes-windows.adb

```diff
@@ -239,56 +244,176 @@ package body Spawn.Processes.Windows is
+                   (Spawn.Windows_API.Pipe_Name_Prefix
+                    & "ada-"
+                    & Process_Id_Image
+                      (Process_Id_Image'First + 1 .. Process_Id_Image'Last)
+                    & '-'
+                    & Pipe_Count_Image
+                      (Pipe_Count_Image'First + 1 .. Pipe_Count_Image'Last));
+               --  Each named pipe is created with unique name.
+
+               Pipe_Count := Pipe_Count + 1;
+
+               Parent_Handle := Windows_API.CreateNamedPipeA
+                 (lpName               => Pipe_Name,
+                  dwOpenMode           =>
+                    Mode (Inbound) + Windows_API.FILE_FLAG_OVERLAPPED,
+                  dwPipeMode           =>
+                    Windows_API.PIPE_TYPE_BYTE
+                      + Windows_API.PIPE_WAIT
+                      + Windows_API.PIPE_REJECT_REMOTE_CLINETS,
+                  nMaxInstances        => 1,
+                  nOutBufferSize       => 0,
+                  nInBufferSize        => 0,
+                  nDefaultTimeOut      => 0,
+                  lpSecurityAttributes => Dont_Inherit_Handle'Access);
+
+               exit when Parent_Handle /= System.Win32.INVALID_HANDLE_VALUE;
+
+               declare
+                  Error : constant Windows_API.DWORD :=
+                    System.Win32.GetLastError;
+
+               begin
+                  Attempts := Attempts - 1;
+
+                  if Error /= Windows_API.ERROR_PIPE_BUSY or Attempts = 0 then
+                     Self.Listener.Error_Occurred (Integer (Error));
+
+                     Interfaces.C.Strings.Free (Pipe_Name);
+
+                     Child_Handle  := System.Win32.INVALID_HANDLE_VALUE;
+                     Parent_Handle := System.Win32.INVALID_HANDLE_VALUE;
+                     Success       := False;
+
+                     return;
+                  end if;
+               end;
+            end;
+         end loop;
+
+         --  Open named pipe to create client handle
 
-         Child_Handle := System.Win32.CreateFileA
-           (lpFileName            => Name'Address,
+         Child_Handle := Windows_API.CreateFileA
+           (lpFileName            => Pipe_Name,
             dwDesiredAccess       => Client_Access (Inbound),
             dwShareMode           => 0,
-            lpSecurityAttributes  => Inherit'Access,
+            lpSecurityAttributes  => Inherit_Handle'Access,
             dwCreationDisposition => System.Win32.OPEN_EXISTING,
-            dwFlagsAndAttributes  => System.Win32.FILE_ATTRIBUTE_NORMAL,
+            dwFlagsAndAttributes  => Windows_API.FILE_FLAG_OVERLAPPED,
             hTemplateFile         => 0);
 
          if Child_Handle = System.Win32.INVALID_HANDLE_VALUE then
             Self.Listener.Error_Occurred (Integer (System.Win32.GetLastError));
-            Success := False;
+
+            Interfaces.C.Strings.Free (Pipe_Name);
+            Ignore (System.Win32.CloseHandle (Parent_Handle));
```

Comment:
I feel like we shouldn't ignore the result here and emit a trace in case it's False

Reply:
It is not necessary here, handle is valid one and not created by some specific functions which requires specific cleanup